src/lib/y2partitioner/actions/create_partition_table.rb

```diff
@@ -96,10 +94,95 @@ def run?
         false
       end
 
-      # Return the device name of the disk
-      # @return [String]
-      def disk_name
-        controller.disk_name
+      # If the operation have destructive consequences, ask for confirmation
+      # to the user.
+      #
+      # @return [Boolean] true if the action is safe or the user confirmed
+      #   to accept the consequences
+      def confirm
+        return true if safe?
+
+        if disk.filesystem
+          confirm_with_filesystem
+        else
+          confirm_with_nested_devices
+        end
+      end
+
+      # Whether performing the action can be considered non-destructive
+      #
+      # @return [Boolean] true if the disk is completely empty and unused or if
+      #   it only contains an empty partition table
+      def safe?
+        desc = disk.descendants.size
+        return true if desc.zero?
+        return false if desc > 1
+
+        disk.partition_table? || disk.encrypted?
```

Comment:
I do not see from method documentation why encrypted disk is fine. I should kill that encrypted device, not?

Reply:
Yes, the definition of "non-destructive" needs to be improved. The idea is that killing an empty partition table or a completely unused encryption layer on the disk (that is, the disk is encrypted but not even formatted) is fine for someone clicking on "Create New Partition Table".

But strictly speaking, both things are "destructive" (they delete something, even if the deleted stuff is not used in any way or shape). So I will improve the documentation.